util/stmtsummary/statement_summary.go

```diff
@@ -52,10 +57,11 @@ type stmtSummaryByDigestKey struct {
 // `prevSQL` is included in the key To distinguish different transactions.
 func (key *stmtSummaryByDigestKey) Hash() []byte {
 	if len(key.hash) == 0 {
-		key.hash = make([]byte, 0, len(key.schemaName)+len(key.digest)+len(key.prevDigest))
+		key.hash = make([]byte, 0, len(key.schemaName)+len(key.digest)+len(key.prevDigest)+len(key.planDigest))
 		key.hash = append(key.hash, hack.Slice(key.digest)...)
 		key.hash = append(key.hash, hack.Slice(key.schemaName)...)
 		key.hash = append(key.hash, hack.Slice(key.prevDigest)...)
+		key.hash = append(key.hash, hack.Slice(key.planDigest)...)
```

Comment:
The same plan but different row count may cause the hash different, is it expected?

Reply:
No, row count is not in normalized plan, so plan digest are the same.
Refer to https://github.com/pingcap/tidb/blob/master/util/plancodec/codec.go#L274

Reply:
Add some tests to cover this case (different plan but the same plan digest).

Reply:
Addressed. Refer to https://github.com/pingcap/tidb/pull/14182/commits/00cbf9ec78a19c55b16c85736e9f6b362436d601